src/libraries/icubmod/usbCamera/linux/V4L_camera.cpp

```diff
@@ -102,6 +109,65 @@ yarp::os::Stamp V4L_camera::getLastInputStamp()
  */
 bool V4L_camera::open(yarp::os::Searchable& config)
 {
+    yarp::os::Value *val;
+    Value* retM;
+    retM=Value::makeList("1.0 0.0 0.0 0.0 1.0 0.0 0.0 0.0 1.0");
+    yDebug()<<"Checking the configuration file";
+    if(config.find("horizontalFov").isNull())
+        configFx=false;
+    if(config.find("verticalFov").isNull())
+        configFy=false;
+    if(config.find("principalPointX").isNull())
+        configPPx=false;
+    if(config.find("principalPointY").isNull())
+        configPPy=false;
+    if(config.find("retificationMatrix").isNull())
+        configRet=false;
+    if(config.find("distortionModel").isNull())
+        configDistM=false;
+    Bottle bt;
+    bt=config.findGroup("cameraDistortionModelGroup");
+    if(bt.isNull())
+        configIntrins=false;
+    else if(bt.find("name").isNull() || bt.find("k1").isNull()
+            || bt.find("k2").isNull() || bt.find("k3").isNull()
+            || bt.find("t1").isNull() || bt.find("t2").isNull())
+        configIntrins=false;
+
+    param.width = config.check("width",yarp::os::Value(320),
+                     "desired width of test image").asInt();
+    param.height = config.check("height",yarp::os::Value(240),
+                     "desired height of test image").asInt();
+    param.horizontalFov=config.check("horizontalFov",Value(0.0),
```

Comment:
`width` and `hight` parameters are already checked in the `fromConfig` function (line 165-272).
That function was meant to collect all the config file parsing into a single place, so I'd suggest to move all the addictional `config.check()` there and remove the double check on `width` and `hight`